pmd-dist/src/main/scripts/run.sh

```diff
@@ -61,7 +61,13 @@ java_heapsize_settings() {
 
 set_lib_dir() {
   if [ -z ${LIB_DIR} ]; then
-    local script_dir=$(dirname "${0}")
+    # Allow for symlinks to this script
+    if [ -L $0 ]; then
+      local script_real_loc=$(readlink "$0")
```

Comment:
Maybe we should `readlink --canonicalize` here to follow links recursively if need be? The script_real_loc would be an absolute path too, which may helps debugging

Reply:
It's not strictly needed (it seems to be working for all cases), however: we could spare the if-statement, because with --canonicalize, readlink always returns the absolute path (whether a symlink has been involved or not).

So, we could just always execute readlink:

`local script_real_loc=$(readlink --canonicalize "$0")`

Reply:
`--canonicalize` is not a valid flag for `readlink` under Mac OS, that's why I went another direction

Reply:
You're right, `--canonicalize` is a GNU extension...